Source/WebKit/UIProcess/API/glib/WebKitWebsiteDataManager.cpp

```diff
@@ -479,6 +490,45 @@ static void webkit_website_data_manager_class_init(WebKitWebsiteDataManagerClass
             nullptr, nullptr,
             FALSE,
             static_cast<GParamFlags>(WEBKIT_PARAM_READWRITE | G_PARAM_CONSTRUCT_ONLY)));
+
+    /**
+     * WebKitWebsiteDataManager:origin-storage-ratio:
+     *
+     * The ratio which is used to calculate the origin data storage based on disk space.
+     * The origin data storage if it is exceeded, request will fail.
+     * This value has to be in range from 0.0 to 1.0.
+     * Value of 0.0 means that data storage is not allowed.
+     * Value out of the range 0.0 - 1.0 means nothing will change.
```

Comment:
Now the documentation says the value has to be in the range 0.0 to 1.0, but in fact it can be -1.0 to 1.0. I don't think this is less confusing.

Reply:
I had to extend the range to give a possibility to set some default value which does not change anything :( I could not use 1.0 as default because in that case the behaviour is different in contrast to not set at all. Do you have any suggestion here?

Reply:
What I'm trying to suggest is that "default value which does not change anything" is meaningless to API users. Who knows what that is? (I don't. :) So forget about it. Always call setOriginQuotaRatio and setTotalQuotaRatio. The only default value should be the one that you pass to g_param_spec_double().

Reply:
(That said, I'd be curious what the internal WebCore defaults are for purposes of choosing what our defaults should be.)

Reply:
OriginQuotaRatio and TotalQuotaRatio are optional. If we always call setOriginQuotaRatio and setTotalQuotaRatio we change the default behavior of QuotaManager(NetworkStorageManager::originStorageManager in case of originQuotaRatio) which will use this value (in contrast to use default quota value (1000MB) or the one set by perOriginStorageQuota)

Reply:
In case of TotalQuotaRatio if we don't set it (the default behaviour) then there won't be any checking and eventual eviction done(NetworkStorageManager::spaceGrantedForOrigin)

Reply:
I'm going to resolve this discussion thread because Carlos has asked you to use -1 for no change after all. I'll drop my complaint.